com.unity.multiplayer.mlapi/Tests/Runtime/RpcQueueTests.cs

```diff
@@ -78,33 +47,158 @@ public IEnumerator RpcQueueUnitTest()
+
+            Assert.IsTrue(testsAreComplete && testsAreValidated);
+
+            //Disable this so it isn't running any longer.
+            rpcPipelineTestComponent.gameObject.SetActive(false);
+
+            yield return null;
+#else
+            yield return null;
+#endif
+        }
 
-            //Stop pinging
-            rpcPipelineTestComponent.PingSelfEnabled = false;
-            Debug.Log("RPC Queue Testing completed.");
+        /// <summary>
+        /// This tests the RPC Queue outbound and inbound buffer capabilities.
+        /// It will send
+        /// </summary>
+        /// <returns>IEnumerator</returns>
+        [UnityTest]
+        public IEnumerator BufferDataValidation()
+        {
+#if UNITY_2020_2_OR_NEWER // Disabling this test on 2019.4 due to ILPP issues on Yamato CI/CD runs
+            Assert.IsTrue(NetworkManagerHelper.StartNetworkManager());
+
+            Guid gameObjectId = NetworkManagerHelper.AddGameNetworkObject("GrowingBufferObject");
+
+            var growingRpcBufferSizeComponent = NetworkManagerHelper.AddComponentToObject<BufferDataValidationComponent>(gameObjectId);
+            NetworkManagerHelper.SpawnNetworkObject(gameObjectId);
+
+            //Start Testing
+            growingRpcBufferSizeComponent.EnableTesting = true;
 
-            //Stop the host
-            NetworkManager.Singleton.StopHost();
+            var testsAreComplete = growingRpcBufferSizeComponent.IsTestComplete();
 
-            //Shutdown the NetworkManager
-            NetworkManager.Singleton.Shutdown();
+            //Wait for the rpc pipeline test to complete or if we exceeded the maximum iterations bail
+            while (!testsAreComplete)
+            {
+                //Wait for 20ms
+                yield return new WaitForSeconds(0.02f);
 
-            Debug.Log($"Exiting status => {nameof(testsAreComplete)}: {testsAreComplete} - {nameof(testsAreValidated)}: {testsAreValidated} - {nameof(instantiatedNetworkManager)}: {instantiatedNetworkManager} - {nameof(exceededMaximumStageIterations)}: {exceededMaximumStageIterations}");
+                testsAreComplete = growingRpcBufferSizeComponent.IsTestComplete();
+            }
 
-            Assert.IsTrue(testsAreComplete && testsAreValidated && instantiatedNetworkManager);
+            //Stop Testing
+            growingRpcBufferSizeComponent.EnableTesting = false;
 
-            GameObject.DestroyImmediate(playerObject);
-            GameObject.DestroyImmediate(networkManagerObject);
+            //Just disable this once we are done.
+            growingRpcBufferSizeComponent.gameObject.SetActive(false);
+
+            Assert.IsTrue(testsAreComplete);
+
+            yield return null;
 #else
             yield return null;
 #endif
+
+        }
+
+
+        [UnityTest]
+        public IEnumerator RpcQueueContainerClass()
```

Comment:
```suggestion
        [Test]
        public void RpcQueueContainerClass()
```

similar to above, you can turn this into a plain test if there is nothing to yield.